### ISSUE TITLE
Deprecation annotation references a no longer existing function in 3.1.0

### ISSUE THREAD

**Issue by StylianosGakis** [apollographql/apollo-kotlin#3915]
https://github.com/apollographql/apollo-kotlin/blob/19af69b6cf3c9126a7bbda6cdee94319acf32629/apollo-api/src/jvmMain/kotlin/com/apollographql/apollo3/api/FileUpload.kt#L15-L16

My guess is that this existed in 3.0.0 but no longer does? When we take the IDE hint and replace with the suggested function it's just red.
It could also be that I am missing some dependency? I don't think so though

**Comment by BoD** [comment]
Woops! You're right, it should reference `DefaultUpload.Builder` instead! Basically what the inside of the method does.

**Comment by StylianosGakis** [comment]
That one no longer has an input for `mimetype` though right?
Not super familiar with these concepts, but is that not needed somehow? Does it automatically get set in some way?

**Comment by BoD** [comment]
> That one no longer has an input for mimetype though right?

Yes, you would need to set the mimetype on the builder, like how it's done [here](https://github.com/apollographql/apollo-kotlin/blob/19af69b6cf3c9126a7bbda6cdee94319acf32629/apollo-api/src/jvmMain/kotlin/com/apollographql/apollo3/api/FileUpload.kt#L22).  It will use `"application/octet-stream"` by default if not supplied. Not 100% certain it matters - probably depends on the server.

**Comment by StylianosGakis** [comment]
Right, so in that case the deprecation replacement should replace this fully with something like
```kotlin
DefaultUpload.Builder()
    .content(file)
    .contentType(mimetype)
    .build()
```
Would this be correct?

**Comment by BoD** [comment]
Exactly 👍

**Comment by StylianosGakis** [comment]
Take a look at the PR whenever you have time, I think it addresses this problem, along with bringing in the correct import

**Comment by martinbonnin** [comment]
Closed in `3.2.0`
